### PR TITLE
Refactor the project and update dependencies

### PR DESCRIPTION
**Pull request by yozhgoor**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ include = ["src/**/*.rs", "README.md", "LICENSE.Apache-2.0", "LICENSE.MIT"]
 
 [dependencies]
 anyhow = "1.0"
-clap = "=3.0.0-beta.5"
+clap = { version = "3.0", features = ["derive"] }
```
Comment (cecton) [inline Cargo.toml:17]: oh it finally released? :grin:

File: src/dependency.rs [hunk 2]
```diff
@@ -1,79 +1,7 @@
-use crate::config::Config;
```
Comment (cecton) [inline src/dependency.rs:1]: Good that you are dropping the lib.rs :+1:

File: src/main.rs [hunk 3]
```diff
@@ -1,9 +1,50 @@
...
 
+mod config;
+mod dependency;
+mod run;
+
+use crate::{
+    config::Config,
+    dependency::{parse_dependency, Dependency},
+    run::execute,
+};
```
Comment (cecton) [inline src/main.rs:13]: I prefer to use `*` to avoid having to maintain a list of imports.

File: src/run.rs [hunk 4]
```diff
@@ -1,11 +1,47 @@
...
+        delete_file,
+        tmp_dir,
+        cli.worktree_branch.clone(),
+        subprocesses,
+    )?;
+
+    match res {
+        Ok(_exit_status) => Ok(()),
+        Err(err) => bail!("problem within the shell process: {}", err),
+    }
```
Comment (cecton) [inline src/run.rs:42]: This should be an `ensure!()`

File: Cargo.toml (hunk 1 above)
Comment (yozhgoor) [reply Cargo.toml:17]: Yup :+1: , at the end of the last year i think

File: src/dependency.rs (hunk 2 above)
Comment (yozhgoor) [reply src/dependency.rs:1]: I hope it will be less confusing

**Review by cecton — APPROVED** [review]

**Pull request merged**